docs/guides/migrating-cra.md

```diff
@@ -0,0 +1,774 @@
+
+- rendering the `App` component in special route file called `root.jsx` in your `appDirectory`
+- moving the other route components into the `routes` directory
+- updating file names to follow our [file name conventions](../api/conventions#file-name-conventions)
+
+So you'll end up with a file structure that looks like this:
+
+```
+app/
+├── routes/
+│   ├── $userId/
+│   │   ├── profile.jsx
+│   │   ├── sales.jsx
+│   └── expenses.jsx
+│   └── index.jsx
+│   └── invoices.jsx
+│   └── $userId.jsx
+└── root.jsx
+```
+
+### The `root` route
+
+The root route (or the "root root" if you're Wes Bos) is responsible for providing the structure of the application. Its default export is a component that renders the full HTML tree.
+
+The index HTML file in a CRA app may look something like this:
+
+```html filename=index.html
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+    <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
+    <meta
+      name="viewport"
+      content="width=device-width, initial-scale=1"
+    />
+    <meta name="theme-color" content="#000000" />
+    <meta
+      name="description"
+      content="Web site created using create-react-app"
+    />
+    <link
+      rel="apple-touch-icon"
+      href="%PUBLIC_URL%/logo192.png"
+    />
+    <link
+      rel="manifest"
+      href="%PUBLIC_URL%/manifest.json"
+    />
+    <title>React App</title>
+  </head>
+  <body>
+    <noscript
+      >You need to enable JavaScript to run this
+      app.</noscript
+    >
+    <div id="root"></div>
+  </body>
+</html>
+```
+
+If you haven't already, create a new file in your `appDirectory` called `root.jsx` and export a component that mirrors that structure:
+
+```js filename=root.jsx
+import * as React from "react";
+
+export default function Root() {
+  return (
+    <html lang="en">
+      <head>
+        <meta charSet="utf-8" />
+        <link rel="icon" href="/favicon.ico" />
+        <meta
+          name="viewport"
+          content="width=device-width, initial-scale=1"
+        />
+        <meta name="theme-color" content="#000000" />
+        <meta
+          name="description"
+          content="Web site created using create-react-app"
+        />
+        <link rel="apple-touch-icon" href="/logo192.png" />
+        <link rel="manifest" href="/manifest.json" />
+        <title>React App</title>
+      </head>
+      <body>
+        <noscript>
+          You need to enable JavaScript to run this app.
+        </noscript>
+        <div id="root"></div>
+      </body>
+    </html>
+  );
+}
+```
+
+Notice that we don't need to use `%PUBLIC_URL%` as our server serves relative paths from the `public` directory.
+
+We can also get rid of the `noscript` tag because we're server rendering now, which means users who disable JavaScript will still be able to see our app (even if they can't interact with it in exactly the same way).
```

Comment:
```suggestion
We can also get rid of the `noscript` tag because we're server rendering now, which means users who disable JavaScript will still be able to see our app and thanks to [progressive enahancement](../pages/philosophy#progressive-enhancement) most of the app will still work anyway.
```

I'm not sure if that URL is correct. I *think* it is.